snapcraft/internal/lifecycle.py

```diff
@@ -252,6 +253,14 @@ def _create_meta(self, step, part_names):
             meta.create_snap_packaging(self.config.data,
                                        self.project_options)
 
+    def _record_snapcraft(self, step, part_names):
```

Comment:
I see it now. I missed it because of the flag. I would just remove it. Or do you have a good reason for it being in meta ?